docs/content/latest/explore/ysql-language-features/data-types.md

```diff
@@ -377,3 +377,59 @@ yugabyte=# SELECT (item).name FROM on_hand WHERE (item).price > 0.99;
+- `tstzrange`, which corresponds to a range of `timestamp with time zone`.
+- `daterange`, which corresponds to a range of `date`.
+- `int4range`, which corresponds to a range of `integer`.
+- `int8range`, which corresponds to a range of `biginteger`.
+- `numrange`, which corresponds to a range of `numeric`.
+- User-defined types.
+
+The following example shows how to provide a range of time for an employee's vacation:
+
+```sql
+CREATE TABLE employees (employee_no int, vacation tsrange);
+
+INSERT INTO employees 
+  VALUES (1227, '[2020-01-01 8:30, 2020-02-02 5:30)');
+```
+
+A non-empty range has a lower bound and an upper bound, with everything between these values included in the range. An inclusive bound indicated by square brackets includes the boundary itself in the range, whereas an exclusive bound indicated by braces excludes the boundary from the range. That is, in the preceding example, the first timestamp is included in the range, and the second timestamp is excluded. If the lower bound is omitted, everything less than the upper bound is included in the range, and if the upper bound is omitted, then everything greater than the lower bound is included in the range. If you omit both bounds, all values of the element type are in the range.
+
+The following is a syntax of an input for a range value, where *empty* is a representation of a range that does not contain anything:
+
+```sql
+(lowerbound, upperbound)
+
+(lowerbound, upperbound]
+
+[lowerbound, upperbound)
+
+[lowerbound, upperbound]
+
+empty
+```
+
+*lowerbound* could be a string of a valid input for the subtype or empty if there is no lower bound. The same logic is applicable to *upperbound*. You can enclose bound values in double quotes, which is a requirement in cases when the value includes parentheses, brackets, commas, double quotes, backslashes. To define an empty-string value, you use `""` (not providing anything is interpreted as defing  an infinite bound). You may use whitespaces before and after values, but not between the parentheses or brackets, as this is interpreted as part of the lower or upper bound value.
+
+You can create a range type using a constructor function named identically to the range type. The constructor functions typically have two or three arguments, with the former constructing a range in standard form (lower bound inclusive, upper bound exclusive), and the latter constructing a range with bounds specified by the third argument (one of strings "`()`", "`(]`", "`[)`", or "`[]`"). The following example shows the constructor with a lower bound, upper bound, and text argument:
+
+```sql
+SELECT numrange(3.0, 10.0, '(]');
+```
+
+In addition to using built-in range types, you can define your own custom ones. The following example shows how to define a range type of subtype `text` and use it in a `SELECT` statement:
+
+```sql
+CREATE TYPE textrange 
+  AS RANGE (subtype = text);
+
+SELECT '( " a " " a ", " z " " z " )'::textrange;
+```
```

Comment:
Looks like we have already a reference page on range types thanks to @bllewell: https://docs.yugabyte.com/latest/api/ysql/datatypes/type_range/

Does it makes sense to add a note somewhere linking to it along the lines of:
```
For more details on range types see ...
```

Reply:
Yes, Bryn sent me that link last night. I'm definitely including it in the doc  (this was the plan all along)